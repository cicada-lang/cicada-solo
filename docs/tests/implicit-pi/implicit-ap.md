# implicit argument insertion when applying implicit function

``` cicada
function id(implicit A: Type, x: A): A {
  return x
}

id(1)
id(implicit Nat, 1)

id("a")
id(implicit String, "a")
```

# multiple implicit arguments

``` cicada
function k(
  implicit A: Type, x: A,
  implicit B: Type, y: B,
): A {
  return x
}

k(123, "abc")
k(implicit Nat, 123, "abc")
k(123, implicit String, "abc")
k(implicit Nat, 123, implicit String, "abc")
```
