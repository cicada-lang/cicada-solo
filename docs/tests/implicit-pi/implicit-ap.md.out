"a": String
"a": String
sole: Trivial
sole: Trivial
sole: Trivial
sole: Trivial